src/main/java/com/danielagapov/spawn/Controllers/BlockedUserController.java

```diff
@@ -0,0 +1,101 @@
+package com.danielagapov.spawn.Controllers;
+
+import com.danielagapov.spawn.DTOs.BlockedUserDTO;
+import com.danielagapov.spawn.Services.BlockedUser.IBlockedUserService;
+import com.danielagapov.spawn.Services.FriendRequest.IFriendRequestService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("api/v1/blocked-users")
+public class BlockedUserController {
+
+    private final IBlockedUserService blockedUserService;
+    private final IFriendRequestService friendRequestService;
+
+    public BlockedUserController(IBlockedUserService blockedUserService, IFriendRequestService friendRequestService) {
+        this.blockedUserService = blockedUserService;
+        this.friendRequestService = friendRequestService;
+    }
+
+    // POST /api/v1/blocked-users/{blockerId}/block/{blockedId}
+    @PostMapping("{blockerId}/block/{blockedId}")
+    public ResponseEntity<Void> blockUser(@PathVariable UUID blockerId,
+                                          @PathVariable UUID blockedId,
+                                          @RequestParam(required = false) String reason) {
```

Comment:
Hey, you could rewrite this for a simpler endpoint by using 2 request params, or a DTO, not two path variables
```suggestion
    // POST /api/v1/blocked-users/block
    @PostMapping("block")
    public ResponseEntity<Void> blockUser(@RequestBody BlockUserCreationDTO) {
```